src/main/java/net/dries007/tfc/common/blocks/devices/BloomeryBlock.java

```diff
@@ -239,5 +261,19 @@ protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockSt
         builder.add(FACING).add(LIT).add(OPEN);
     }
 
+    //todo: revise this to work properly, since this is a total guess (even if by some miracle this is the right idea, check to see if the directions are right)
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter getter, BlockPos pos, CollisionContext context)
+    {
+        return switch (state.getValue(FACING))
+            {
+                case NORTH -> BASE_SHAPES[0];
+                case SOUTH -> BASE_SHAPES[1];
+                case WEST -> BASE_SHAPES[2];
+                case EAST -> BASE_SHAPES[3];
+                default -> Shapes.block();
```

Comment:
No reason for these to be an array.
Also by default throw an illegal argument exception instead of handling cases that should never happen